client/flags.go

```diff
@@ -58,14 +67,57 @@ func PostCommands(cmds ...*cobra.Command) []*cobra.Command {
 		c.Flags().String(FlagChainID, "", "Chain ID of tendermint node")
 		c.Flags().String(FlagNode, "tcp://localhost:26657", "<host>:<port> to tendermint rpc interface for this chain")
 		c.Flags().Bool(FlagUseLedger, false, "Use a connected Ledger device")
-		c.Flags().Int64(FlagGas, DefaultGasLimit, "gas limit to set per-transaction; set to 0 to calculate required gas automatically")
 		c.Flags().Float64(FlagGasAdjustment, DefaultGasAdjustment, "adjustment factor to be multiplied against the estimate returned by the tx simulation; if the gas limit is set manually this flag is ignored ")
 		c.Flags().Bool(FlagAsync, false, "broadcast transactions asynchronously")
 		c.Flags().Bool(FlagJson, false, "return output in json format")
 		c.Flags().Bool(FlagPrintResponse, true, "return tx response (only works with async = false)")
 		c.Flags().Bool(FlagTrustNode, true, "Don't verify proofs for query responses")
 		c.Flags().Bool(FlagDryRun, false, "ignore the --gas flag and perform a simulation of a transaction, but don't broadcast it")
 		c.Flags().Bool(FlagGenerateOnly, false, "build an unsigned transaction and write it to STDOUT")
+		// --gas can accept integers and "simulate"
+		c.Flags().Var(&GasFlagVar, "gas", fmt.Sprintf(
+			"gas limit to set per-transaction; set to %q to calculate required gas automatically (default %d)", GasFlagSimulateString, DefaultGasLimit))
 	}
 	return cmds
 }
+
+// Gas flag parsing functions
+
+// GasSetting encapsulates the possible values passed through the --gas flag.
+type GasSetting struct {
+	Simulate bool
+	Gas      int64
```

Comment:
Can we change the type to `uint64` ?

Reply:
Yes, I believe since gas is always monotonically increasing, it can be unsigned?

Reply:
This should not be done in this pr. There have been many discussions about this, the latest is happening in the Chris' security issue.

Reply:
Well that's settled 👍

Reply:
ok @alessio let's leave it for later then

Reply:
> This should not be done in this pr. There have been many discussions about this, the latest is happening in the Chris' security issue.

@ValarDragon Issue/PR number please?

Reply:
@fedekunze TODO comment added

Reply:
#2173

Reply:
yeah I doubt we want uints based on Jae's comment in ^